src/modules/commander/Commander.cpp

```diff
@@ -809,7 +809,11 @@ Commander::handle_command(vehicle_status_s *status_local, const vehicle_command_
 
 				if (!enforce_in_air && !land_detector.landed && !is_ground_rover(&status)) {
 					if (cmd_arms) {
-						mavlink_log_critical(&mavlink_log_pub, "Arming denied! Not landed");
+						if (armed_local->armed) {
```

Comment:
You could also just accept the command in that case (it's technically not denied, it just has no effect)

Reply:
That would also be an option. I chose to send a warning so that we don't have this silently happen and we don't detect that we're doing something wrong.